Drivers/MotorChannel/Inc/ZP_D_PWMChannel.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef ZP_D_PWM_CHANNEL_HPP_
+#define ZP_D_PWM_CHANNEL_HPP_
+
+#include "main.h"
+
+#include "ZP_D_MotorChannel.hpp"
+
+class PWMChannel : public MotorChannel {
+    public:
+        PWMChannel(uint16_t pin_num, GPIO_TypeDef* port, TIM_HandleTypeDef* timer,
+					uint16_t timer_channel,uint32_t clock_frequency);
+
+        void setup(void);
+        void set(uint8_t percent);
+    private:
+        uint16_t pin_num_;
+        GPIO_TypeDef* port_;
+        TIM_HandleTypeDef *timer_;     //handle to the timer
+        uint16_t timer_channel_;       //channel of the timer
+        uint32_t clock_frequency_;     //system clock frequency
+        uint32_t period_ticks_;
+
+        //values in microseconds(us)
+        static constexpr uint32_t PWM_PERIOD = 20000;
```

Comment:
what is the difference between period_ticks_ and PWM_PERIOD? And why is PWM_PERIOD set to 20000?

Reply:
Sorry, I realize I should've looked closer at these. Since they were in the original LOS driver, I left them in.

I'm not sure how we should set the prescaler and period, as that would be two unknowns when all we have is the desired frequency and the system clock frequency. I assumed that PWM_PERIOD would just be our general counts amount. 

I will remove the redundancy, retest, and repush

Reply:
Or should it just take the counts period from whatever is already set on that timer?

Reply:
So your correct, if we want the feature where at startup, no matter the input frequency we can calculate the prescaler/period, that would be two unknowns. So that means we need to have one hardcoded, prop the period?? And at that point we would just solve for the prescaler and set both.